packages/react/src/components/Popover/index.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { useState, useEffect, ReactNode, forwardRef, Ref } from 'react';
+import { createPortal } from 'react-dom';
+import { useId } from 'react-id-generator';
+import { Placement } from '@popperjs/core';
+import { usePopper } from 'react-popper';
+import { isBrowser } from '../../utils/is-browser';
+import { Cauldron } from '../../types';
+
+import classnames from 'classnames';
+import ClickOutsideListener from '../ClickOutsideListener';
+import Button from '../Button';
+import FocusTrap from 'focus-trap-react';
+import focusableSelector from '../../utils/focusable-selector';
+import AriaIsolate from '../../utils/aria-isolate';
+import useSharedRef from '../../utils/useSharedRef';
+
+export type PopoverVariant = 'alert' | 'custom';
+
+type BaseProps = React.HTMLAttributes<HTMLDivElement> & {
+  target: React.RefObject<HTMLElement> | HTMLElement;
+  variant?: PopoverVariant;
+  show: boolean;
+  onClose: () => void;
+  placement?: Placement;
+  portal?: React.RefObject<HTMLElement> | HTMLElement;
+};
+
+type CustomProps = BaseProps & {
+  variant: 'custom';
+  applyButtonText?: string;
+  onApply?: () => void;
+  closeButtonText?: string;
+  infoText?: string;
+  chidlren: ReactNode;
+} & Cauldron.LabelProps;
+
+type AlertProps = BaseProps & {
```

Comment:
`infoTextId` needs to be included in props here.

Reply:
Do we allow to pass `infoTextId` from outside of the component?
We generate id by ourself in main component, 
that's why i added type on a component level instead of the type?

Reply:
Ah, I missed that it was set below.

Since we don't export the props including it here would be a best practice since an author would still see that prop set via typescript. We can leave it as is for now.